voxblox/src/integrator/esdf_integrator.cc

```diff
@@ -199,19 +199,30 @@ void EsdfIntegrator::updateFromTsdfBlocks(const BlockIndexList& tsdf_blocks,
       } else {
         // If this voxel DID exist before.
         // There are three main options:
+        // (1a): if was fixed before but not anymore, raise.
         // (1) lower: esdf or tsdf is fixed, and tsdf is closer to surface than
         // it used to be.
         // (2) raise: esdf or tsdf is fixed, and tsdf is further from surface
         // than it used to be.
         // (3) sign flip: tsdf and esdf have different signs, otherwise the
         // lower and raise rules apply as above.
         if (tsdf_fixed || esdf_voxel.fixed) {
-          if ((esdf_voxel.distance > 0 &&
-               tsdf_voxel.distance + config_.min_diff_m <
-                   esdf_voxel.distance) ||
-              (esdf_voxel.distance <= 0 &&
-               tsdf_voxel.distance - config_.min_diff_m >
-                   esdf_voxel.distance)) {
+          if (!tsdf_fixed && esdf_voxel.fixed) {
+            // New case: have to raise the voxel
+            esdf_voxel.distance =
+                signum(tsdf_voxel.distance) * config_.default_distance_m;
+            esdf_voxel.parent.setZero();
+            esdf_voxel.fixed = false;
+            raise_.push(global_index);
+            esdf_voxel.in_queue = true;
+            open_.push(global_index, esdf_voxel.distance);
+            num_raise++;
+          } else if ((esdf_voxel.distance > 0 &&
```

Comment:
`static_cast<FloatingPoint>(0.0)` ?

Reply:
Usually no need to cast explicit values. ;) 0.0f does the same thing, changed this.